api/api/controllers/search_controller.py

```diff
@@ -541,7 +544,8 @@ def related_media(uuid: str, index: str, filter_dead: bool) -> list[Hit]:
     # Exclude the current item and mature content.
     s = s.query(related_query & ~Term(identifier=uuid) & ~Term(mature=True))
     # Exclude the dynamically disabled sources.
-    s = _exclude_filtered(s)
+    excluded_providers = get_dynamically_excluded_providers()
+    s = s.exclude(Terms(provider=excluded_providers))
```

Comment:
Shouldn't the excluded providers _always_ be excluded? Would it make sense to add this to the base `create_search_query` function?

Reply:
Yes, they should always be excluded. We shouldn't add any provider to `excluded` providers if its media should not be excluded.  Thank you for suggestion!